### PR TITLE
odo-init-image test automation for PSI

### PR DESCRIPTION
**Pull request by anandrkskd**
This PR is adding scripts for automation for tests against PSI cluster.
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

solves https://github.com/openshift/odo/issues/4315

### CODE REVIEW

**Comment by dharmit** [comment]
Needs https://github.com/openshift/odo/pull/4946 to go in first.

File: scripts/one_time_configure_cluster.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (prietyc123) [inline scripts/one_time_configure_cluster.sh:1]: Did you already run this script on the related VM?

File: scripts/openshiftci-test.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# fail if some commands fails
+set -e 
+
+# hide commands that deals with secrets
+set +x
+if [[ -f $ODO_RABBITMQ_AMQP_URL ]]; then
+    export AMQP_URI=$(cat $ODO_RABBITMQ_AMQP_URL)
+fi
+export AMQP_URI=${AMQP_URI:?"Please set AMQP_URI env with amqp uri or provide path of file containf it as ODO_RABBITMQ_AMQP_URL env"}
```
Comment (prietyc123) [inline scripts/openshiftci-test.sh]: *of file contains?

File: scripts/openshiftci-test.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+export SENDQUEUE=amqp.ci.queue.init.image.send
+export SENDTOPIC=amqp.ci.topic.init.image.send
+export SENDEXCHANGE=amqp.ci.exchange.init.image.send
+export RUNSCRIPT=srcipts/testing.sh
```
Comment (prietyc123) [inline scripts/openshiftci-test.sh]: typo with `scripts`

File: scripts/one_time_configure_cluster.sh (hunk 1 above)
Comment (anandrkskd) [reply scripts/one_time_configure_cluster.sh:1]: This script is related to configuring cluster with permissions that is required for developer to access registry and exposing the registry route.

File: scripts/testing.sh [hunk 4]
```diff
@@ -1,12 +1,41 @@
...
+
+#----
+
+# execute tests
+sh -c ". ./scripts/setup_script_e2e.sh && . ./scripts/run_script_e2e.sh"
+
+#cleanup
+popd
+rm -rf $RANDOM_DIR
+oc delete project $IMAGE_NAMESPACE
```
Comment (prietyc123) [inline scripts/testing.sh:43]: If scripts error out in between this namespace will also left out in the cluster. Don't you think we should add same label `odo` here as well.

**Review by prietyc123 — CHANGES_REQUESTED** [review]

**Comment by prietyc123** [comment]
@anandrkskd Aren't we planning to push generated reports to `artifacts_dir` . Similar to https://github.com/openshift/odo/blob/main/scripts/openshiftci-presubmit-all-tests.sh#L71

File: scripts/testing.sh (hunk 4 above)
Comment (anandrkskd) [reply scripts/testing.sh:43]: yes, I will add labels to this namespace too.
Thank you for pointing this out.

**Comment by anandrkskd** [comment]
Fixes https://github.com/openshift/odo/issues/4315

**Comment by anandrkskd** [comment]
Closing this as we will be using github-actions for the tests with PR https://github.com/redhat-developer/odo-init-image/pull/102
/close

**Pull request closed without merging**